src/lib.rs

```diff
@@ -48,6 +48,7 @@ pub mod int;
     all(target_arch = "x86_64", target_os = "none"),
     all(target_arch = "x86_64", target_os = "uefi"),
     all(target_arch = "arm", target_os = "none"),
+    all(target_arch = "riscv32", target_os = "none"),
```

Comment:
I think we should also gate on `not(target_feature = "f")`

Reply:
huh, my initial thinking was that if there are gates on non-fpu on relevant elements inside math.rs it is ok not to gate it on module inclusion

I'll do whatever you guys tell me, just created this PR to speed things up and have the thing going on somewhere since `esp-hal` was clearly not a correct place for it

Reply:
> A fair amount of the intrinsics in libm don't have native instructions even on cpu's that do support floating point operations.

Is there any point in gating the math module at all then? Presumably, if the backend supports code-generation instead of using an intrinsic it will just do that, and not emit a call to the intrinsic. Compile times I suppose?

Reply:
> Is there any point in gating the math module at all then?

Yes, we must not include it when there is already a libm implementation providing them. This libm implementation is generally part of the system libc. We are allowlisting targets without libc here.

Reply:
Ok, so I brought back the original commit